cmd/query/app/server.go

```diff
@@ -344,16 +344,42 @@ func (s *Server) Start() error {
 	return nil
 }
 
-// Close stops http, GRPC servers and closes the port listener.
+// Close stops HTTP, GRPC servers and closes the port listener.
 func (s *Server) Close() error {
 	var errs []error
-	errs = append(errs, s.queryOptions.TLSGRPC.Close())
-	errs = append(errs, s.queryOptions.TLSHTTP.Close())
+
+	if err := s.queryOptions.TLSGRPC.Close(); err != nil && !isConnAlreadyClosedError(err) {
+		errs = append(errs, fmt.Errorf("failed to close TLS gRPC server: %w", err))
+	}
+	time.Sleep(200 * time.Millisecond)
+
+	if err := s.queryOptions.TLSHTTP.Close(); err != nil && !isConnAlreadyClosedError(err) {
+		errs = append(errs, fmt.Errorf("failed to close TLS HTTP server: %w", err))
+	}
+	time.Sleep(200 * time.Millisecond)
```

Comment:
none of this is necessary. I already explained that TLSxxx.Close has nothing to do with servers or with the error in question, it just stops the file watchers.

Reply:
Sorry for the confusion. I was just trying to be extra cautious in ensuring all resources were being closed properly. Based on your feedback, I've removed the unnecessary `TLSGRPC.Close()` and `TLSHTTP.Close()` operations.